app/views/contexts/list_context.py

```diff
@@ -95,12 +96,17 @@ def _build_list_items_context(
 
             yield list_item_context
 
-    def _get_item_title(self, summary_definition, list_item_id, is_primary):
+    def _get_item_title(
+        self,
+        summary_definition: Mapping[str, Any],
+        list_item_id: Optional[str],
+        is_primary: bool,
+    ) -> Any:
         rendered_summary = self._placeholder_renderer.render(
             summary_definition, list_item_id
         )
-
+        # Type :ignore as mapping would be str at this point
         if is_primary:
-            rendered_summary["item_title"] += lazy_gettext(" (You)")
+            rendered_summary["item_title"] += lazy_gettext(" (You)")  # type: ignore
```

Comment:
This is a valid error. This errors because `self._placeholder_renderer.render` return a Mapping which is immutable and then mypy error because we are mutating it. It should be updated to return a `dict`.

Reply:
I am not sure if we can do that...
As:
`self._placeholder_renderer.render` returns `QuestionnaireSchema.get_mutable_deepcopy(dict_to_render)` which returns `data` which is `Any`.
Not sure if we can change it to dict

Reply:
👍  Marked `self._placeholder_renderer.render` return type as dict. Still `rendered_summary["item_title"] += lazy_gettext(" (You)")` will return Any

Reply:
👍